Client/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift

```diff
@@ -99,7 +100,28 @@ extension BrowserViewController: TopToolbarDelegate {
   }
 
   func topToolbarDidPressReload(_ topToolbar: TopToolbarView) {
-    tabManager.selectedTab?.reload()
+    if WalletFeatureFlags.SNSDomainResolverEnabled, let url = topToolbar.currentURL, url.schemelessAbsoluteDisplayString.endsWithSupportedSNSExtension, let currentStatus = Preferences.Wallet.Web3DomainOption(rawValue: Preferences.Wallet.resolveSNSDomainNames.value) {
+      switch currentStatus {
+      case .ask:
+        // show name service interstitial page
+        showSNSDomainInterstitialPage(originalURL: url, visitType: nil)
+      case .enabled:
+        Task { @MainActor in
+          let isPrivateMode = PrivateBrowsingManager.shared.isPrivateBrowsing
+          if let rpcService = BraveWallet.JsonRpcServiceFactory.get(privateMode: isPrivateMode) {
+            let (resolvedUrl, status, _) = await rpcService.snsResolveHost(url.absoluteString)
+            if let resolvedUrl = resolvedUrl, status == .success {
+              tabManager.selectedTab?.loadRequest(URLRequest(url: resolvedUrl))
+            }
+          }
+          tabManager.selectedTab?.loadRequest(URLRequest(url: url))
```

Comment:
This will not work.
First it will do line `114` and right after that, it will fall through to line `117` because there is no return statement after line `114`. So it is double loading.